src/Controls/src/Core/Element/Element.cs

```diff
@@ -386,9 +396,9 @@ public Element Parent
 
 		void SetParent(Element value)
 		{
-			Element realParent = RealParent;
+			Element currentParent = ParentOverride ?? GetRealParent(false);
```

Comment:
It still feels like we're duplicating logic here and on the Parent.get call.

Can you consollidate the logic into a "GetParent(bool logWarningIfParentHasBeenCollected)" method?